src/eos-updater-apply.c

```diff
@@ -120,42 +161,74 @@ apply (GTask *task,
     {
       g_autofree gchar *rev = NULL;
 
-      if (!ostree_repo_resolve_rev (repo, orig_refspec, TRUE, &rev, &error))
-        goto error;
+      if (!ostree_repo_resolve_rev (repo, orig_refspec, TRUE, &rev, error))
+        return FALSE;
 
       if (rev)
         {
-          if (!ostree_repo_prepare_transaction (repo, NULL, cancel, &error))
-            goto error;
+          if (!ostree_repo_prepare_transaction (repo, NULL, cancel, error))
+            return FALSE;
 
           ostree_repo_transaction_set_refspec (repo, orig_refspec, NULL);
 
-          if (!ostree_repo_commit_transaction (repo, NULL, cancel, &error))
-            goto error;
+          if (!ostree_repo_commit_transaction (repo, NULL, cancel, error))
+            return FALSE;
         }
     }
 
   if (!ostree_sysroot_simple_write_deployment (sysroot,
-                                               NULL,
+                                               osname,
                                                new_deployment,
-                                               merge_deployment,
+                                               booted_deployment,
                                                0,
                                                cancel,
-                                               &error))
-    goto error;
+                                               error))
+    return FALSE;
 
   newbootver = ostree_deployment_get_deployserial (new_deployment);
 
-  g_task_return_boolean (task, bootversion != newbootver);
-  goto cleanup;
+  if (!eos_extensions_save (data->extensions,
+                            repo,
+                            cancel,
+                            error))
+    return FALSE;
+  g_clear_object (&data->extensions);
+
+  head_commit_timestamp = g_date_time_new_from_unix_utc (ostree_commit_get_timestamp (update_commit));
+
+  if (!eos_avahi_generate_service_file (repo,
+                                        head_commit_timestamp,
+                                        error))
+    return FALSE;
```

Comment:
I need to think about this some more, but I am wondering if using an Avahi `.service` file is still actually the best option. There are no parts of the updater which run continuously apart from Avahi — `eos-autoupdater` runs on a timer, `eos-updater` only runs when the state machine changes state, and `eos-update-server` only runs when socket-activated on an incoming HTTP connection.

Unless we could use path-based activation (`man systemd.path`) to run a little service whenever `/ostree/repo/refs/remotes` changes, which would update the Avahi `.service` file then exit?